### PR TITLE
Add benchmarking/fix some performance issues

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/background_process/__init__.py [hunk 1]
```diff
@@ -11,7 +12,7 @@
     reset_comms,
 )
 
-IPC_ADDRESS = ("localhost", 9898)  # Specify the IP address and port
+IPC_ADDRESS = "/tmp/aikido_socket.sock"  # Specify the IP address and port
```
Comment (willem-delbare) [inline aikido_firewall/background_process/__init__.py]: php gebruikt dezelfde potentieel, kan je die aikido_socket_python noemen?
also, kan je eventueel de gebruiker met een env var
AIKIDO_TMP_DIR laten kiezne?

File: aikido_firewall/sources/flask.py [hunk 2]
```diff
@@ -10,34 +10,46 @@
...
+
+        req = request_ctx.request
+        # Set body :
+        context = get_current_context()
+        if context:
+            if req.is_json:
+                context.body = req.get_json()
+                context.set_as_current_context()
+            else:
+                context.body = req.form
```
Comment (willem-delbare) [inline aikido_firewall/sources/flask.py:33]: what if xml? maak je jira?

Comment (bitterpanda63) [reply aikido_firewall/sources/flask.py:33]: Yes we can test that, I'll make a ticket

File: aikido_firewall/background_process/__init__.py (hunk 1 above)
Comment (bitterpanda63) [reply aikido_firewall/background_process/__init__.py]: Done

File: benchmarks/flask-mysql-benchmarks.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        'status is 200': (r) => r.status === 200,
+    });
+    return res.timings.duration; // Return the duration of the request
+}
+function route_test(amount, route, method="GET", data=default_payload) {
+    let time_in_mss_8086 = []
+    let time_in_mss_8087 = []
+    for (let i = 0; i < amount; i++) {
+        time_in_mss_8086.push(measureRequest(BASE_URL_8086 + route, method, data));
+        time_in_mss_8087.push(measureRequest(BASE_URL_8087 + route, method, data));
```
Comment (hansott) [inline benchmarks/flask-mysql-benchmarks.js]: k6 can calculate much of these stats out of the box 😬 

Not sure why we used this manual approach

Comment (hansott) [reply benchmarks/flask-mysql-benchmarks.js]: Have a look at https://k6.io/docs/javascript-api/k6-metrics/trend/

File: docs/socket.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Socket
+
+- Add ref here to `AIKIDO_TMP_DIR`
```
Comment (hansott) [inline docs/socket.md]: Can we expand this a bit? or if people are likely not going to set this (and we just use it for testing) then maybe not document?

File: sample-apps/django-mysql-gunicorn/.env.example [hunk 5]
```diff
@@ -7,3 +7,5 @@ MYSQL_ROOT_PASSWORD="password"
 # Aikido keys
 AIKIDO_DEBUG=true
 AIKIDO_TOKEN="AIK_secret_token"
+DOCKER_PORT=8082
+FIREWALL_DISABLED=0
```
Comment (hansott) [inline sample-apps/django-mysql-gunicorn/.env.example]: I think this is going to be confusing for people using the reference as a guide to install firewall in their app

maybe leave this away? if the env var is not set, firewall is still enabled?

File: sample-apps/flask-mysql/app.py [hunk 6]
```diff
@@ -20,10 +24,6 @@
 
 @app.route("/")
 def homepage():
-    aikido_firewall.set_user({
-        "id": 1,
-        "name": "Wout"
-    })
```
Comment (hansott) [inline sample-apps/flask-mysql/app.py:26]: why remove?

File: docs/socket.md (hunk 4 above)
Comment (bitterpanda63) [reply docs/socket.md]: Euh maybe let's just make this a JIRA issue and remove it for now, dunno yet exactly where we should put this in the docs. We should also explain how the whole agent thing works

File: sample-apps/flask-mysql/app.py (hunk 6 above)
Comment (bitterpanda63) [reply sample-apps/flask-mysql/app.py:26]: This should be called in middleware, not when handling the request

File: sample-apps/django-mysql-gunicorn/.env.example (hunk 5 above)
Comment (bitterpanda63) [reply sample-apps/django-mysql-gunicorn/.env.example]: I already changed how this works for flask-mysql, forgot to port to other apps

File: sample-apps/django-mysql-gunicorn/docker-compose.benchmark.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/django-mysql-gunicorn/docker-compose.benchmark.yml]:
```suggestion

```

File: sample-apps/django-mysql/docker-compose.benchmark.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/django-mysql/docker-compose.benchmark.yml]:
```suggestion

```

File: sample-apps/flask-mongo/docker-compose.benchmark.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/flask-mongo/docker-compose.benchmark.yml]:
```suggestion

```

File: sample-apps/flask-mysql-uwsgi/docker-compose.benchmark.yml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/flask-mysql-uwsgi/docker-compose.benchmark.yml]:
```suggestion

```

File: sample-apps/django-mysql-gunicorn/docker-compose.benchmark.yml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on:
+      db:
+        condition: service_healthy
+    extra_hosts:
+      - "app.local.aikido.io:host-gateway"
+    environment:
+      - FIREWALL_DISABLED=1
+  backend:
+    environment:
+      - AIKIDO_TOKEN="test_aikido_token
```
Comment (hansott) [inline sample-apps/django-mysql-gunicorn/docker-compose.benchmark.yml]:
```suggestion
      - AIKIDO_TOKEN="test_aikido_token"
```

File: sample-apps/django-mysql/docker-compose.benchmark.yml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+      - .env
+    depends_on:
+      - db
+    extra_hosts:
+      - "app.local.aikido.io:host-gateway"
+    environment:
+      - FIREWALL_DISABLED=1
+  backend:
+    environment:
+      - AIKIDO_TOKEN="test_aikido_token
```
Comment (hansott) [inline sample-apps/django-mysql/docker-compose.benchmark.yml]:
```suggestion
      - AIKIDO_TOKEN="test_aikido_token"
```

File: sample-apps/flask-mongo/docker-compose.benchmark.yml [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+      - db
+    networks:
+      - default_network
+    extra_hosts:
+      - "app.local.aikido.io:host-gateway"
+    environment:
+      - FIREWALL_DISABLED=1
+  backend:
+    environment:
+      - AIKIDO_TOKEN="test_aikido_token
```
Comment (hansott) [inline sample-apps/flask-mongo/docker-compose.benchmark.yml]:
```suggestion
      - AIKIDO_TOKEN="test_aikido_token"
```

File: sample-apps/flask-mysql/docker-compose.benchmark.yml [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+      - "8087:5000"
+    depends_on:
+      - db
+    extra_hosts:
+      - "app.local.aikido.io:host-gateway"
+    environment:
+      - FIREWALL_DISABLED=1
+  backend:
+    environment:
+      - AIKIDO_TOKEN="test_aikido_token
```
Comment (hansott) [inline sample-apps/flask-mysql/docker-compose.benchmark.yml]:
```suggestion
      - AIKIDO_TOKEN="test_aikido_token"
```

File: sample-apps/flask-postgres/docker-compose.benchmark.yml [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+      - db
+    networks:
+      - default_network
+    extra_hosts:
+      - "app.local.aikido.io:host-gateway"
+    environment:
+      - FIREWALL_DISABLED=1
+  backend:
+    environment:
+      - AIKIDO_TOKEN="test_aikido_token
```
Comment (hansott) [inline sample-apps/flask-postgres/docker-compose.benchmark.yml]:
```suggestion
      - AIKIDO_TOKEN="test_aikido_token"
```

File: sample-apps/flask-postgres/docker-compose.benchmark.yml [hunk 16]
```diff
@@ -0,0 +1,24 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/flask-postgres/docker-compose.benchmark.yml]:
```suggestion

```

File: sample-apps/flask-mysql/docker-compose.benchmark.yml [hunk 17]
```diff
@@ -0,0 +1,22 @@
+# docker-compose.prod.yml
```
Comment (hansott) [inline sample-apps/flask-mysql/docker-compose.benchmark.yml]:
```suggestion

```

**Pull request merged**